### PR TITLE
Forecast: Dark mode UI improvements and improved links

### LINKED ISSUE #3150: Forecast: UI in dark mode inconsistent with the theme
### Description

The hover state of the forecast makes URLs blue. That makes sense when the theme is the default one, but in dark mode, it should just underline it(similar to what regular search links look like).

I'll be making a PR fixing this in a few hours.

## Steps to recreate

- Change theme to dark
- Search "weather"
- Hover over links

## People to notify

@himanshu0113 

---

Instant Answer Page: https://duck.co/ia/view/forecast

### LINKED ISSUE #3152: Forecast: Mobile links don't work
### Description

Mobile links for future forecasts don't go to [DarkSky](http://darksky.net), instead they seem to redirect to the page itself. The link in the location works however.

## Steps to recreate

- Use a mobile device
- Search `weather`
- Click link in "Mon"/"Fri"/"Sat" etc.

## People to notify

@himanshu0113

---

Instant Answer Page: https://duck.co/ia/view/forecast

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
## Description of new Instant Answer, or changes

- Fixed all issues in dark mode, made appropriate changes to code organisation.
- DarkSky provides forecast details of a place on a exact date. Switched links in the days view with that instead of the generic meta link. Makes the links more appropriate. Ex. [Tomorrow's weather forecast for Phoenixville, PA](https://darksky.net/details/40.1304,-75.5149/2017-02-23/). This also fixes the issue with getting mobile links working.

## Related Issues and Discussions

Fixes #3152 

## People to notify

@bsstoner @himanshu0113

---

Instant Answer Page: https://duck.co/ia/view/forecast

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/forecast/forecast.css [hunk 1]
```diff
@@ -274,3 +272,20 @@
 #fe_temp_switch .is-active {
 	font-weight: bold;
 }
+
+/*
+    Dark mode specific changes
+*/
+
+.dark-bg .fe_label a:hover {
+    color: #dadada;
```
Comment (moollaza) [inline share/spice/forecast/forecast.css]: We shouldn't hard-code this, we need to fix this at a higher level as it affects all IA links in dark mode -- I'll open a task for the team.

File: share/spice/forecast/forecast.css [hunk 2]
```diff
@@ -274,3 +272,20 @@
...
+    text-decoration: underline;
+}
+
+.dark-bg a#fe_temp_switch:hover {
+    color: #c2c2c2;
+}
+
+.dark-bg .fe_icon { display: none; }
```
Comment (moollaza) [inline share/spice/forecast/forecast.css]: This doesn't seem necessary? I always see the correct icons in Chrome/FF/Safari. Which browser are you using?

Comment (moollaza) [inline share/spice/forecast/forecast.css]: Ah I see you're moving these. The `.dark-header` class is still present on the page in the dark theme -- this does't seem necessary. Please remove this

Comment (nalinbhardwaj) [inline share/spice/forecast/forecast.css]: Isn't it better to have all changes for dark mode in a seperate block? Having them inline makes maintenance tougher.

File: share/spice/forecast/forecast.css (hunk 1 above)
Comment (nalinbhardwaj) [inline share/spice/forecast/forecast.css]: Oh okay. I had previously done a similar change for the timer IA #3134 so I thought this was necessary. I'll remove these.

Comment (nalinbhardwaj) [inline share/spice/forecast/forecast.css]: @moollaza removed this. Note that #3150 shouldn't be closed until the team fixes this for everyone :)

File: share/spice/forecast/forecast.css (hunk 2 above)
Comment (nalinbhardwaj) [inline share/spice/forecast/forecast.css]: @moollaza what do you think about this? I have made the other change. If you feel that having a separate block is unnecessary I'll drop this change...

**Comment by nalinbhardwaj** [comment]
@duckduckgo/community-leaders Anyone up for reviewing this? I have some more improvements to the forecast IA, so I'm waiting for this to be merged first.

**Comment by gautamkrishnar** [comment]
:+1:

**Comment by moollaza** [comment]
@nalinbhardwaj Thanks! This LGTM :+1:

**Pull request merged**